src/internal/tab-bar.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {html, css, LitElement, customElement} from 'lit-element';
+
+import type {PlaygroundInternalTab} from './tab.js';
+
+/**
+ * A horizontal bar of tabs.
+ *
+ * Slots:
+ * - default: The <playground-internal-tab> tabs.
+ */
+@customElement('playground-internal-tab-bar')
+export class PlaygroundInternalTabBar extends LitElement {
+  static styles = css`
+    :host {
+      display: flex;
+      overflow-x: auto;
+    }
+
+    :host::-webkit-scrollbar {
+      display: none;
+    }
+
+    div {
+      display: flex;
+    }
+  `;
+
+  /**
+   * Get or set the active tab.
+   */
+  get active(): PlaygroundInternalTab | undefined {
+    return this._active;
+  }
+
+  set active(tab: PlaygroundInternalTab | undefined) {
+    /**
+     * Note the active tab can be set either by setting the bar's `active`
+     * property to the tab, or by setting the tab's `active` property to
+     * true. The two become synchronized according to the following flow:
+     *
+     *   bar click/keydown
+     *           |
+     *           v
+     *   bar.active = tab ---> changed? ---> tab.active = true
+     *           ^                                 |
+     *           |                                 v
+     *   bar tabchange listener        changed from false to true?
+     *           ^                                 |
+     *           |                                 |
+     *           +--- tab dispatches tabchange <---+
+     */
+    const oldActive = this._active;
+    if (tab === oldActive) {
+      return;
+    }
+    this._active = tab;
+    if (oldActive !== undefined) {
+      oldActive.active = false;
+    }
+    if (tab !== undefined) {
+      tab.active = true;
+    } else {
+      // Usually the tab itself emits the tabchange event, but we need to handle
+      // the "no active tab" case here.
+      this.dispatchEvent(
+        new CustomEvent<{tab?: PlaygroundInternalTab}>('tabchange', {
+          detail: {tab: undefined},
+          bubbles: true,
+        })
+      );
+    }
+  }
+
+  private _tabs: PlaygroundInternalTab[] = [];
+  private _active: PlaygroundInternalTab | undefined = undefined;
+
+  render() {
+    return html`
+      <div role="tablist">
+        <slot
+          @slotchange=${this._onSlotchange}
+          @click=${this._activateTab}
+          @keydown=${this._onKeydown}
+          @tabchange=${this._activateTab}
+        ></slot>
+      </div>
+    `;
+  }
+
+  private _onSlotchange(event: Event) {
+    this._tabs = (
+      event.target as HTMLSlotElement
+    ).assignedElements() as PlaygroundInternalTab[];
```

Comment:
How "custom" do you want this to be? like only for your use case? might be a problem if someone tries to use it and places another element as a child? Might be good to filter? In MWC list we assume the interface of any component with the attribute `[mwc-list-item]` is `ListItem` and filter out other things like list separators.

Close comment if you don't expect to use this outside of the bounds of PgE's.

If you do decide to go with this, make sure you change references to `localName` below

Reply:
well we don't have a use for it now, so I think it's fine